packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -437,6 +439,12 @@ def handle_error(self, args: ErrorArgs):
 
         return self.run_ai_cell(cell_args, prompt)
 
+    def _append_exchange(self, prompt: str, output: str):
+        """Appends a conversational exchange between user and an OpenAI Chat
+        model to a transcript that will be included in future exchanges."""
+        self.transcript.append(HumanMessage(prompt))
+        self.transcript.append(AIMessage(output))
+
```

Comment:
There should be bounds on the length of the transcript passed to the LLM, since some LLMs have shorter token limits. Can you mimic the implementation of the history in chat, where we only allow up to 2 Human-AI exchanges? That would involve modifying this method to remove all but the last Human-AI exchange before appending the new Human-AI exchange to `self.transcript`.

If you wish to implement longer history as well, you can make the size of the history configurable via `%ai config`, while defaulting to 2 exchanges. See this PR for a reference: https://github.com/jupyterlab/jupyter-ai/pull/962

Reply:
@dlqqq I assume you mean `%config AiMagics.<option> = <value>`, not `%ai config`? At least that's what [current documentation says](/jupyterlab/jupyter-ai/blob/main/docs/source/users/index.md#configuring-a-default-model).

I'll look into that!